### ISSUE TITLE
What is the ground truth ATE in the data?

### ISSUE THREAD

**Issue by sheryc** [rpryzant/causal-bert-pytorch#3]
Hi Raid! I'd like to know what the real ATE is in the data? Thank you!

**Comment by rpryzant** [comment]
Thanks for reaching out! 
Right after reading in the data:
https://github.com/rpryzant/causal-bert-pytorch/blob/master/CausalBert.py#L305

You can calculate the raw unadjusted ATE with this code: 

```
def ATE_unadjusted(T, Y):
    x = defaultdict(list)
    for t, y in zip(T, Y):
        x[t].append(y)
    T0 = np.mean(x[0])
    T1 = np.mean(x[1])
    return T0 - T1
 ```
    
  and you can set the args to `df['T']` and `df['Y']`

**Comment by sheryc** [comment]
Thank you! This is the unadjusted ATE but I wonder if the provided testdata is semi-synthetic. And should the args be `df['T']` and `df['Y']`? Is the ATE calculation actually `T1 - T0`? Thanks!

**Comment by rpryzant** [comment]
Oops - yes! You are right re: inputs. Sorry for the confusion. It should be `df['T']` and `df['Y']`

Yes the testdata is semi-synthetic. The Y's were the only part of the data that were simulated I believe, and the simulation code can be found here: 
https://github.com/rpryzant/causal-text/blob/main/src/simulation.py#L193

See section 6.1 for details on the simulation:
https://nlp.stanford.edu/pubs/pryzant2021causal.pdf

The unadjusted ATE (what I gave you) is P(Y | T = 0) - P(Y | T = 1). The confound-adjusted ATE is sum_C { P(Y | T = 0, C) - P(Y | T = 1, C) }. Like this: 

```
def ATE_adjusted(C, T, Y):
    x = defaultdict(list)
    for c, t, y in zip(C, T, Y):
        x[c, t].append(y)

    C0_ATE = np.mean(x[0,0]) - np.mean(x[0,1])
    C1_ATE = np.mean(x[1,0]) - np.mean(x[1,1])
    return np.mean([C0_ATE, C1_ATE])
```

Note that the simulation code I linked too wasn't the same version of the code that was used to generate `testdata.csv`, though, so there might be some differences.....so sorry!
